### ISSUE TITLE
charon crash when starting on boot and using Linux CLOCK_MONOTONIC source.

### ISSUE THREAD

**Issue by simonjhaggett** [strongswan/strongswan#435]
**System (please complete the following information):**
 - OS: Linux (Yocto Poky v3.1)
 - Kernel version (if applicable): 5.4
 - strongSwan version(s): 5.9.1
 - Tested/confirmed with the latest version: no

**Describe the bug**
charon crashes with a divide error if started within 1 second of system uptime, when the Linux CLOCK_MONOTONIC source is used.

**To Reproduce**
1. Ensure charon is built to use the Linux CLOCK_MONOTONIC source (the default as I understand). This clock starts at system boot time (see `Documentation/core-api/timekeeping.rst` in the Linux 5.4 source tree).
2. Start charon on boot, within one second of the kernel starting.

**Expected behavior**
charon starts successfully.

**Logs/Backtraces**
charon crashes, producing a kernel log similar to the following:
```
traps: charon[427] trap divide error ip:7fd81d4fbf60 sp:7ffc877c2a10 error:0 in libcharon.so.0.0.0[7fd81d4de000+62000]
```

**Additional context**
I've traced the crash to `receiver_create()` (src/libcharon/network/receiver.c). This calls `time_monotonic(NULL)`, which returns 0 in this case (i.e. the monotonic clock source reports 0 seconds). I've added custom logging in my builds to confirm this. This then leads to a divide error when the following expression is evaluated (as `now` is 0):
```
.secret_offset = random() % now,
```
I can  avoid this crash with the following change to force `secret_offset` to 0 if `now` is 0:
```
.secret_offset = (now == 0) ? 0 : random() % now,
```

However, I'm not sure if having `secret_offset` set to 0 then negates its purpose ("hides our system time"). I also see that `secret_offset` can be set to 0 if the monotonic clock source reports a non-zero but small seconds value (e.g. 1). I'm using strongSwan 5.9.1, but the code involved appears to be the same on master.

Is having `secret_offset` set to 0 a problem? If not, and my change above is acceptable, then I'd be happy to raise a PR for this change. Otherwise, might a bigger change be needed here?

**Comment by tobiasbrunner** [comment]
> However, I'm not sure if having `secret_offset` set to 0 then negates its purpose ("hides our system time").

I guess it does, but it's not the "real" system time anyway with monotonic time like that, only the uptime. Not sure if that's a problem.

Theoretically, we could use an actual random value as base and change the calculations (while properly handling over-/underflows, so that might be prone to errors) or just mask the timestamp in the COOKIE by XORing it with a random value (not really much of a protection). We also could avoid timestamps altogether and use an identifier for the secret that's used (doesn't work with the current code that switches secrets based on use count, but in another branch I've already implemented a time-based switch of those secrets, where that might work, although cookies would be valid longer than they are now).

For now, the simplest fix is to just avoid the crash. I've pushed that to the _435-secret-offset_ branch.
